package/gluon-mesh-olsrd/luasrc/lib/gluon/upgrade/360-gluon-mesh-olsrd-setup-intf

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local site = require 'gluon.site'
+local util = require 'gluon.util'
+local wireless = require 'gluon.wireless'
+local sysconfig = require 'gluon.sysconfig'
+local util = require 'gluon.util'
+local olsrd = require 'gluon.olsrd'
+
+local mesh_interfaces = util.get_role_interfaces(uci, 'mesh')
+local uplink_interfaces = util.get_role_interfaces(uci, 'uplink')
+local client_interfaces = util.get_role_interfaces(uci, 'client')
+
+local mesh_interfaces_uplink = {}
+local mesh_interfaces_client = {}
+local mesh_interfaces_other = {}
+for _, iface in ipairs(mesh_interfaces) do
+	if util.contains(uplink_interfaces, iface) then
+		table.insert(mesh_interfaces_uplink, iface)
+	elseif util.contains(client_interfaces, iface) then
+		table.insert(mesh_interfaces_client, iface)
+	else
+		table.insert(mesh_interfaces_other, iface)
+	end
+end
+
+
+local intf = {
+	wired_mesh = {},
+	vpn_mesh = {},
+	radio_mesh = {},
+}
+
+intf.all_intfs = {}
+
+for _, l in ipairs({ intf.wired_mesh, intf.vpn_mesh, intf.radio_mesh }) do
+	for _, n in ipairs(l) do
+		table.insert(intf.all_intfs, n)
+	end
+end
+
+-- get all mesh radios and mesh lans and then add them to olsrd
+wireless.foreach_radio(uci, function(radio, index, config)
+	local radio_name = radio['.name']
+	table.insert(intf.radio_mesh, 'mesh_' .. radio_name)
+	if uci:get('network', 'ibss_' .. radio_name, 'proto') then
+		table.insert(intf.radio_mesh, 'ibss_' .. radio_name)
+	end
+end)
+
+local vpn_mesh = false
+
+if pcall(function() require 'gluon.mesh-vpn' end) then
+	local vpn_core = require 'gluon.mesh-vpn'
+
+	if vpn_core.enabled() then
+		-- mesh_vpn is a interface that has the right ifname
+		-- we can't use mesh-vpn (dash instead of underscore) since it's not a uci interface
+		table.insert(intf.vpn_mesh, 'mesh_vpn')
+		vpn_mesh = true
+	end
+end
+
+local function has_role_mesh(ifname)
+	local roles = uci:get('gluon', 'iface_' .. ifname, 'role')
+
+	local has_role = false
+
+	if roles then
+		for _, r in ipairs(roles) do
+			if r == 'mesh' then
+				has_role = true
+			end
+		end
+	end
+
+	return has_role
+end
+
+table.insert(intf.wired_mesh, 'loopback')
+
+local has_uplink_mesh = false
+local has_other_mesh = false
+
+for _,i in pairs(mesh_interfaces) do
+	if util.contains(uplink_interfaces, i) then
+		has_uplink_mesh = true
+	else
+		has_other_mesh = true
+	end
+end
+
+if has_uplink_mesh then
+	-- use uplink with macvlan so we do not have traffic leaks
```

Comment:
this fixes the problem that happens when meshing on wan and having mesh vpn enabled, for ex. this might make sense for all of gluon. thoughts?

Reply:
The macvlan part is interesting, but I think it should be removed for now, too. If there is a need to use mesh-vpn + wan without vxlan, then this should be a separate pull-request. And should ideally behave the same and work both for babel and olsr2. With emphasizes on "if", I think @NeoRaider was not very keen on supporting mesh-on-wan/lan without vxlan.

Reply:
it has been moved to the graz branch, but I'd like to still know the opinion of @NeoRaider on this, since it seems useful